### ISSUE TITLE
missing separators for some steps

### ISSUE THREAD

**Issue by arctic-penguin** [topgrade-rs/topgrade#167]
For example:
* [npm](https://github.com/topgrade-rs/topgrade/blob/6a6a84b0c511158841a731dda79e4aa4fb0689cd/src/steps/node.rs#L218..L222)
* [pnpm](https://github.com/topgrade-rs/topgrade/blob/6a6a84b0c511158841a731dda79e4aa4fb0689cd/src/steps/node.rs#L232..L236)

When these steps are executed, there isn't necessarily an indication something gets executed. In my case the previous step was `pip`, which ran successfully, then topgrade asked me if I wanted to try again, and the desktop notification indicated that the step `pnpm` had failed.

## Additional Details
- Which operating system or Linux distribution are you using? Arch
- How did you install Topgrade? `git clone; cargo run`
- Which version are you running? HEAD

**Comment by 9999years** [comment]
Hmm, the npm and pnpm steps call `NPM::upgrade`, which does print a separator:

https://github.com/topgrade-rs/topgrade/blob/1b49b70833acf4f9d3d6e80d921eb90e6d2d1ae1/src/steps/node.rs#L99

<details>
<summary>Example output on my machine (on the <code>dev</code> branch), showing the separators</summary>

```ShellSession
$ cargo run -- --only node
   Compiling topgrade v10.1.2 (/Users/wiggles/topgrade)
    Finished dev [unoptimized + debuginfo] target(s) in 3.31s
     Running `target/debug/topgrade --only node`

―― 10:18:39 - Node Package Manager ―――――――――――――――――――――――――――――――――――――――――――――

removed 13 packages, changed 75 packages, and audited 224 packages in 2s

14 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

―― 10:18:41 - Yarn Package Manager ―――――――――――――――――――――――――――――――――――――――――――――
yarn global v1.22.19
[1/4] 🔍  Resolving packages...
warning create-react-app > tar-pack > tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Rebuilding all packages...

success Saved lockfile.
success Saved 54 new dependencies.
info Direct dependencies
└─ create-react-app@5.0.1
info All dependencies
├─ ansi-styles@4.3.0
...
└─ yallist@4.0.0
✨  Done in 1.34s.

―― 10:18:43 - PNPM ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Already up to date
Done in 134ms

―― 10:18:43 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
npm: OK
yarn: OK
pnpm: OK
```

</details>

Do you have log output or anything? Definitely curious about this.

**Comment by arctic-penguin** [comment]
I had an issue with pnpm and npm on my machine which I have now ironed out. Cannot reproduce anymore. Thanks for the quick reply!

**Comment by 9999years** [comment]
I have two fixes for this! One, in #170, moves the separators to the correct place, so that the banner will be printed before the command that failed. The second, in #171, prints detailed error messages when commands fail; this would have made it clear exactly what program failed and what it had to say, even before running in debug-mode.

**Comment by 9999years** [comment]
I think we can close this now that #170 and #171 are merged.
